### PR TITLE
Report errors if any tuned kernels fail to generate or compile

### PR DESCRIPTION
**Pull request by AlexBrownAMD**
Rocblas does not report errors at build time if a tuned kernel fails to generate or compile. Lack of error reporting is caused by enabling either merge files or lazy-loading features. The problem is current error checking is based on number of files generated, and when files are merged expected output is always 1 unless all kernels fail to build.

This change updates the code to error out if any kernels fail to generate or compile. Error checking can be disabled by setting the errorTolerant option. Using the library to do things like test or benchmark will run in error tolerant mode by default. Error messages are normally only enabled when building kernels from a previously generated library logic file (like during a rocblas build).

Tested locally using valid logic files and ones with errors introduced and produces anticipated results.

Also tested a full rocblas build and it detected an invalid tuned kernel - I'll work on an update to the logic files that will need to be merged into rocblas before this change is used.

### CODE REVIEW

File: Tensile/TensileCreateLibrary.py
Comment (AlexBrownAMD) [reply Tensile/TensileCreateLibrary.py]: Updated

File: Tensile/TensileCreateLibrary.py
Comment (AlexBrownAMD) [reply Tensile/TensileCreateLibrary.py]: Updated

**Comment by AlexBrownAMD** [comment]
> Please check if Integration test fail is related to your change or not.

Seeing those integration errors happen on mainline develop locally as well. Not related to this change, but I'll track down the source of those errors now.

**Comment by nakajee** [comment]
> > Please check if Integration test fail is related to your change or not.
> 
> Seeing those integration errors happen on mainline develop locally as well. Not related to this change, but I'll track down the source of those errors now.

Really?
From the error message, I feel it is related to your change.
  KeyError: 'SolutionIndex'
I cannot find the same error in other PRs or CI on develop branch.

**Comment by AlexBrownAMD** [comment]
> > > Please check if Integration test fail is related to your change or not.
> > 
> > 
> > Seeing those integration errors happen on mainline develop locally as well. Not related to this change, but I'll track down the source of those errors now.
> 
> Really? From the error message, I feel it is related to your change. KeyError: 'SolutionIndex' I cannot find the same error in other PRs or CI on develop branch.

Appears to be caused by rocBLAS PR #1369. The build pulled by the integration test must have been updated to include that change set. The tests are failing because they can't find the logic files in rocBLAS since they've been moved

**Comment by nakajee** [comment]
> Appears to be caused by rocBLAS PR #1369. The build pulled by the integration test must have been updated to include that change set. The tests are failing because they can't find the logic files in rocBLAS since they've been moved

OK. It is a bit old one. 
Why we do not see the same error in other Tensile PRs?
If we are sure the error is not caused by this PR, I am OK with the current change.

**Comment by AlexBrownAMD** [comment]
> > Appears to be caused by rocBLAS PR #1369. The build pulled by the integration test must have been updated to include that change set. The tests are failing because they can't find the logic files in rocBLAS since they've been moved
> 
> OK. It is a bit old one. Why we do not see the same error in other Tensile PRs? If we are sure the error is not caused by this PR, I am OK with the current change.

The rocblas build may have been updated really recently (the build pulled by integration tests doesn't track very quickly with changes on the github repo). If other PRs reran the integration tests now then the bug would show up there too. Bug is easy to reproduce locally on plain develop branch. Just posted PR #1632 to fix the integration tests.

**Comment by nakajee** [comment]
> The rocblas build may have been updated really recently (the build pulled by integration tests doesn't track very quickly with changes on the github repo). If other PRs reran the integration tests now then the bug would show up there too. Bug is easy to reproduce locally on plain develop branch. Just posted PR #1632 to fix the integration tests.

OK. I approved it. Let's see if it can fix the error.

**Comment by nakajee** [comment]
Please merge the latest commit here.

**Comment by nakajee** [comment]
Thanks for your update.
Integration test looks OK now.
However, I can see some pre_checkin tests are failing.
Can you check if it is caused by your change or not?

[gw0] [  0%] FAILED Tensile/Tests/yaml_only/test_config.py::test_config[Tensile/Tests/pre_checkin/dgemm_asm.yaml] 

[gw1] [  6%] FAILED Tensile/Tests/yaml_only/test_config.py::test_config[Tensile/Tests/pre_checkin/dgemm_general_batch_asm.yaml] 

[gw3] [  8%] FAILED Tensile/Tests/yaml_only/test_config.py::test_config[Tensile/Tests/pre_checkin/hgemm_general_batch_hpa_asm_nn.yaml] 

[gw1] [  9%] FAILED Tensile/Tests/yaml_only/test_config.py::test_config[Tensile/Tests/pre_checkin/hgemm_hpa_asm_f32_alphabeta_nn.yaml]

**Review by babakpst — DISMISSED** [review]
Blocked until after FC merge.

**Review by nakajee — APPROVED** [review]
The previous CI fail seems to be OK now.
Please make sure gfx908 extended fail is not caused by this change.

**Pull request merged**
